bcachefs/bcachefs.c

```diff
@@ -413,6 +496,7 @@ int Bcachefs_close(Bcachefs *this)
 int Bcachefs_iter(const Bcachefs *this, Bcachefs_iterator *iter, enum btree_id type)
 {
     iter->type = type;
+    iter->jset_entry = NULL; 
```

Comment:
Memory leaks are a pain to debug. The more we discuss the more I believe the initialization should be done when the pointer is allocated and I see the following alternatives:

* init to `{0}` to be done outside of the func or create an init func for the iterator
  * optionaly add a `memcmp()` to make sure the pointer is clean
* change the signature of the func to return the pointer to the iterator which would allow us to `malloc` and set to `{0}`

Reply:
@Delaunay , do we let @obilaniu cut for this discussion? ;)